### PR TITLE
Added Aarch64 support to oclmath library

### PR DESCRIPTION
**Pull request by lrpablo**
Added support for Aarch64

### CODE REVIEW

File: oclmath/fpcontrol.h [hunk 1]
```diff
@@ -62,8 +72,10 @@
         _mm_setcsr( *mode );
 #elif defined( __PPC__)
         fpu_control = *mode;
-#elif defined (__arm__)
+#elif defined (__arm__)
```
Comment (bader) [inline oclmath/fpcontrol.h:75]:
```suggestion
#elif defined (__arm__)
```
Please, no trailing whitespaces.

File: oclmath/reference_math.cpp [hunk 2]
```diff
@@ -4800,7 +4800,7 @@ static long double reference_scalblnl(long double x, long n)
 
     return x;
     
-#elif defined(__arm__) // ARM .. sizeof(long double) == sizeof(double) 
+#elif defined(__arm__)  || defined( __aarch64__ )// ARM .. sizeof(long double) == sizeof(double)
```
Comment (bader) [inline oclmath/reference_math.cpp:4803]:
```suggestion
#elif defined(__arm__)  || defined( __aarch64__ ) // ARM .. sizeof(long double) == sizeof(double) 
```

File: oclmath/rounding_mode.cpp [hunk 3]
```diff
@@ -230,7 +238,7 @@ void UnFlushToZero( void *p)
         flags &= ~_FPU_MASK_NI;
         _FPU_SETCW(flags);
         #else
-        #error Unknown arch
+        #error Unknown arch 1
```
Comment (bader) [inline oclmath/rounding_mode.cpp:241]: What is the value of this change?

File: oclmath/rounding_mode.cpp [hunk 4]
```diff
@@ -220,6 +224,10 @@ void UnFlushToZero( void *p)
         #define _ARM_FE_FTZ     0x1000000
 		#define _ARM_FE_NFTZ    0x0
         #define _FPU_SETCW(cw) __asm__ ("VMSR FPSCR,%0" : :"ri" (cw))
+    #elif defined(__aarch64__)
+        #define _ARM_FE_FTZ     0x1000000
+		#define _ARM_FE_NFTZ    0x0
+        #define _FPU_SETCW(cw) __asm__ ("MSR FPCR,%0" : :"ri" (cw))
```
Comment (bader) [inline oclmath/rounding_mode.cpp:230]:
```suggestion
        #define _ARM_FE_FTZ     0x1000000
    		#define _ARM_FE_NFTZ    0x0
        #define _FPU_SETCW(cw) __asm__ ("MSR FPCR,%0" : :"ri" (cw))
```

NOTE: your change invalidates the branch for `__arm__` case by removing
```c++
		static const int ftz_modes[ kRoundingModeCount ] = { _ARM_FE_FTZ, _ARM_FE_NFTZ };
		const int *f = ftz_modes;
		_FPU_SETCW( f[1] );
```

The same applies to `FlushToZero` function.

**Review by bader — CHANGES_REQUESTED** [review]

**Comment by bader** [comment]
@lrpablo, we already have https://github.com/KhronosGroup/SYCL-CTS/pull/890, which is supposed to do the same improvement. 
@hvdijk, FYI.

**Comment by lrpablo** [comment]
Okay, I hadn't seen that. My apologies for the inconvenience. Do you have an estimated date for when it will be present in the release?

Regards and thank you,

________________________________

File: oclmath/rounding_mode.cpp (hunk 3 above)
Comment (lrpablo) [reply oclmath/rounding_mode.cpp]: Should not have been there

**Comment by bader** [comment]
> Do you have an estimated date for when it will be present in the release?

SYCL-CTS project doesn't have releases, so you can use the head of the SYCL-2020 branch to validate your favorite SYCL implementation.
With regards to https://github.com/KhronosGroup/SYCL-CTS/pull/890, PRs are merged once they are approved and pass pre-commit validation. I'll try to facilitate the code review process.

**Pull request closed without merging**